locale.c

```diff
@@ -6300,28 +6300,37 @@ S_my_langinfo_i(pTHX_
         utf8ness_t is_utf8 = UTF8NESS_UNKNOWN;
         char * scratch_buf = NULL;
         Size_t scratch_buf_size = 0;
+        const U8 * first_variant;
+        Size_t len;
```

Comment:
I think these belong within the loops that use them.

There's no harm in having separate declarations of `len` within each loop, and it makes it easier to reason about the use of `len`, since it would not available outside those inner blocks.

Similarly for `first_variant`.

I don't see any uses of `len` or `first_variant` outside of their respective loops, or in the rest of the commit series, but I may have missed them.

I believe this would also allow `len` to be const.

Reply:
It's because not both are loops, so the 2nd  does conflict with the first.

But I'm not happy with the duplicated logic; it was ok before these commits, as it was just a line or two.  But it has expanded.  I've now pushed a revised version which has one loop that iterates over all of the tested strings.  It makes a portion more complicated, but the actual logic is squashed into one instance